File: python/programmers/heap/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while scoville[0] < K:
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        result = first + (second * 2)
        heapq.heappush(scoville, result)
        answer += 1
        if len(scoville) == 1 and scoville[0] < K:
            return -1
    
    return answer

File: java/src/main/java/ps_traning/programmers/level_2/더_맵게.java
package ps_traning.programmers.level_2;

import java.util.PriorityQueue;

public class 더_맵게 {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int s : scoville) pq.offer(s);
        while (pq.size() >= 2 && pq.peek() < K) {
            int first = pq.poll();
            int second = pq.poll();
            pq.offer(first + second * 2);
            answer++;
        }
        if (pq.peek() < K) answer = -1;
        return answer;
    }
}
